Libraries/SignalingClient/src/peer_connection_client.cpp

```diff
@@ -303,16 +334,42 @@ bool PeerConnectionClient::Shutdown()
 		control_socket_->Close();
 	}
 
+	if (capacity_socket_.get() != nullptr)
+	{
+		capacity_socket_->Close();
+	}
+
+	bool shouldFireSignal = false;
+
+	if (state_ == CONNECTED)
+	{
+		shouldFireSignal = true;
+	}
+
 	state_ = NOT_CONNECTED;
 
+	if (shouldFireSignal)
+	{
+		SignalDisconnected.emit();
+	}
+

```

Comment:
no, because there are other valid states in which this can be called that should also be overridden.

 ie: `RESOLVING` => `NOT_CONNECTED` is a valid transition that should occur as a result of this code

Reply:
I see, this should work right?

	if (state_ == CONNECTED)
	{
		state_ = NOT_CONNECTED;
		SignalDisconnected.emit();
	}
	else
	{       
		state_ = NOT_CONNECTED;
	}

Reply:
yes that would work...what is it about the curernt solution that you dislike?

Reply:
Oh, it just confuses me when I first read it.